includes/core/server/request/class-generic.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Generic.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Exceptions\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Exceptions\Server_Request_Exception;
+use WCPay\Core\Server\Request;
+use Requests;
+
+/**
+ * Generic WCPay Server Request.
+ *
+ * This class is not extendable, if you need something specific,
+ * it will not be generic anymore, create your own.
+ */
+final class Generic extends Request {
```

Comment:
It's great that we have this class for quick requests to the server. Though, I fail to see (haven't run the code) how this class is locked to extending.